patches/server/0714-Add-Feature-Generation-API.patch

```diff
@@ -121,3 +142,257 @@ index 46ce973ca68420825316352ad4df907edf52bfec..65e9ecda0df8a22f93f5979a8c9ac74e
      }
  
      @Override
+diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaBlockState.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaBlockState.java
+index bfdeb4f2971e65c38334f2f90c66ef62564e83e6..5ec7a3903838ce2f60926782965107e84b44643c 100644
+--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaBlockState.java
++++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaBlockState.java
+@@ -271,13 +271,18 @@ public class CraftMetaBlockState extends CraftMetaItem implements BlockStateMeta
+         return this.blockEntityTag != null;
+     }
+ 
++    // Paper start - Delegate to utility method
+     @Override
+     public BlockState getBlockState() {
+-        Material stateMaterial = (this.material != Material.SHIELD) ? this.material : CraftMetaBlockState.shieldToBannerHack(this.blockEntityTag); // Only actually used for jigsaws
+-        if (this.blockEntityTag != null) {
+-            switch (this.material) {
++        return createBlockState(this.material, this.blockEntityTag);
++    }
++
++    public static BlockState createBlockState(Material material, CompoundTag blockEntityTag) {
++        Material stateMaterial = (material != Material.SHIELD) ? material : CraftMetaBlockState.shieldToBannerHack(blockEntityTag); // Only actually used for jigsaws
++        if (blockEntityTag != null) {
++            switch (material) {
+                 case SHIELD:
+-                    this.blockEntityTag.putString("id", "banner");
++                    blockEntityTag.putString("id", "banner");
```

Comment:
why have all these this qualifiers been nuked?

Reply:
The method has been made static